src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,32 @@
+import { ConfigService } from '@nestjs/config';
+import { UnauthorizedException } from '@nestjs/common';
+import { User } from './user.entity';
+import { JwtPayload } from './jwt_payload.interface';
+import { UserRepository } from './user.repository';
+import { PassportStrategy } from '@nestjs/passport';
+import { InjectRepository } from '@nestjs/typeorm';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+
+export class JwtStrategy extends PassportStrategy(Strategy) {
+  constructor(
+    @InjectRepository(UserRepository)
+    private usersRepository: UserRepository,
+    private configService: ConfigService
+  ) {
+    super({
+      secretOrKey: configService.get('JWT_SECRET'),
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken()
+    });
+  }
+
+  async validate(payload: JwtPayload): Promise<User> {
+    const { email } = payload;
+    const user = await this.usersRepository.findOne({ where: { email } });
```

Comment:
What if the transaction fails? should you have a try/catch block to handle this scenario?